File: frontend/src/app/services/usersbloodrequest.service.ts
import { Injectable } from '@angular/core';
import { Usersbloodrequest } from '../dao/usersbloodrequest';
import { HttpClient } from '@angular/common/http';
import { BloodbankRequest } from '../dto/bloodbank-request';

@Injectable({
  providedIn: 'root'
})
export class UsersbloodrequestService {
 
private baseUrl ="http://localhost:9090"

constructor(private httpClient:HttpClient) { }

  getBloodRequestById(userid: number) {
    return this.httpClient.get<Usersbloodrequest[]>(`${this.baseUrl}/getBloodRequestById/${userid}`)
  }
  acceptRequest(requestid: number, acceptoruserid: number, acceptorrole: String) {
   const acceptRequest={
      acceptoruserid:acceptoruserid,
      acceptorrole:acceptorrole
    }
    return this.httpClient.put<any>(`${this.baseUrl}/addAcceptRequest/${requestid}`,acceptRequest)
  }
  getAllBloodRequests() {
    return this.httpClient.get<Usersbloodrequest[]>(`${this.baseUrl}/getAllUsersBloodRequest`)
  }

  

  saveBloodRequest(usersBloodRequest: Usersbloodrequest) {
    return this.httpClient.post<any>(`${this.baseUrl}/saveUsersBloodRequest`,usersBloodRequest)
  }

  ReceivedBlood(requestid: number, status:string) {
    return this.httpClient.put<any>(`${this.baseUrl}/receivedBlood/${requestid}`,status)
    
  }
  notReceivedBlood(requestid: number, status: string) {
    return this.httpClient.put<any>(`${this.baseUrl}/notReceivedBlood/${requestid}`,status)
  }
}

File: backend/src/main/java/bloodbankapplication/service/UsersBloodRequestService.java
package bloodbankapplication.service;

import java.time.LocalDate;
import java.util.List;

import bloodbankapplication.dao.UsersBloodRequest;

public interface UsersBloodRequestService {

	public void saveUsersBloodRequest(UsersBloodRequest usersBloodRequest);

	public List<UsersBloodRequest> getAllUsersBloodRequest();

	public void addAcceptRequest(Integer requestid, int acceptoruserid, String acceptorrole,String fullfillmentstatus);

	public List<UsersBloodRequest> getBloodRequestByUserId(Integer userid);

	public void receivedBlood(Integer requestid, String status, LocalDate fullfilleddate,
			String fullfillmentstatus);

	public void notReceivedBlood(Integer requestid, String status, int acceptoruserid,
			String acceptorrole, String fullfillmentstatus);

}
